### ISSUE TITLE
[Screen reader] Use checkboxes for the facets instead of links

### ISSUE THREAD

**Issue by christinach** [pulibrary/orangelight#4896]
### Steps to recreate
Use the mobile screen reader or voice over on a mac.
Find the facets (button) and search for resources in the Library. 
The screen reader reads that the facet is a link. 
The page will render and will read a lot of information.
 
### Issues
- Confusing for non sighted screen reader users that cannot have the whole picture and set up of the UI at a glance.
- Tester expected to see checkboxes that will keep you on the page as you select them. 
- Link implies that will redirect the user to a new page and makes the tester hesitant to use it.
- It leaves the screen and makes it difficult to go back and forth in the facets

### Severity
High

### User impact
- Confusing for non sighted screen reader users that cannot have the whole picture and set up of the UI at a glance.
- Tester expected to see checkboxes that will keep you on the page as you select them. 
- Link implies that will redirect the user to a new page and makes the tester hesitant to use it.
- It leaves the screen and makes it difficult to go back and forth in the facets

### Recommended fix
- Option 1: use checkboxes and a submit button. (submit filter)
- Option 2: ..

### Acceptance criteria
- [ ] Facets are checkboxes and not links.
- [ ] I can successfully use this feature with Voice Over in Safari.
- [ ] I have checked this feature using one of the following browser extensions and confirmed that it does not add any level A or level AA issues:
    * axe dev tools
    * Lighthouse
    * Wave
- [ ] ...

### Notes
This issue also came up in our first catalog fable test with another screen reader user.
Related to https://github.com/pulibrary/orangelight/issues/4888
